File: backend/src/routes/curriculum.js
// ============================================
// Müfredat (Curriculum) API Routes
// GET /api/curriculum/:examType - Sınav türüne göre dersler + konular
// GET /api/curriculum/subjects/:subjectId/topics - Ders konuları
// ============================================

import { Router } from 'express';
import { PrismaClient } from '@prisma/client';

const router = Router();
const prisma = new PrismaClient();

// Sınav türüne göre tüm dersleri ve konularını getir
// examType: LGS | TYT | AYT
router.get('/:examType', async (req, res) => {
  try {
    const { examType } = req.params;

    const subjects = await prisma.subject.findMany({
      where: { examType: examType.toUpperCase() },
      include: {
        topics: { orderBy: { id: 'asc' } }
      },
      orderBy: { displayOrder: 'asc' }
    });

    res.json(subjects);
  } catch (error) {
    console.error('Müfredat hatası:', error);
    res.status(500).json({ error: error.message });
  }
});

// Belirli bir dersin konularını getir
router.get('/subjects/:subjectId/topics', async (req, res) => {
  try {
    const topics = await prisma.topic.findMany({
      where: { subjectId: parseInt(req.params.subjectId) },
      orderBy: { id: 'asc' }
    });

    res.json(topics);
  } catch (error) {
    console.error('Konu listesi hatası:', error);
    res.status(500).json({ error: error.message });
  }
});

export default router;

File: backend/app/routes/curriculum.py
"""
Müfredat (Ders & Konu) API Route'ları
"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session, joinedload
from app.database import get_db
from app.models import Subject

router = APIRouter(prefix="/api/curriculum", tags=["Müfredat"])


def subject_to_dict(s):
    return {
        "id": s.id,
        "name": s.name,
        "examType": s.exam_type,
        "totalQuestions": s.total_questions,
        "coefficient": s.coefficient,
        "displayOrder": s.display_order,
        "topics": [
            {"id": t.id, "name": t.name, "subjectId": t.subject_id}
            for t in (s.topics or [])
        ],
    }


@router.get("/{exam_type}")
def get_curriculum(exam_type: str, db: Session = Depends(get_db)):
    """Sınav türüne göre ders ve konuları getir"""
    subjects = (
        db.query(Subject)
        .options(joinedload(Subject.topics))
        .filter(Subject.exam_type == exam_type.upper())
        .order_by(Subject.display_order)
        .all()
    )
    return [subject_to_dict(s) for s in subjects]
